zerver/webhooks/github/headers.py

```diff
@@ -0,0 +1,55 @@
+push_headers = {"HTTP_X_GITHUB_EVENT": "push"}
+pull_request_headers = {"HTTP_X_GITHUB_EVENT": "pull_request"}
+
+fixtures_to_headers_map = {
+    "assigned_pull_request": pull_request_headers,
+    "check_run_completed": {"HTTP_X_GITHUB_EVENT": "check_run"},
+    "check_run_in_progress": {"HTTP_X_GITHUB_EVENT": "check_run"},
+    "closed_pull_request": pull_request_headers,
+    "commit_comment": {"HTTP_X_GITHUB_EVENT": "commit_comment"},
+    "create": {"HTTP_X_GITHUB_EVENT": "create"},
+    "delete": {"HTTP_X_GITHUB_EVENT": "delete"},
+    "deployment": {"HTTP_X_GITHUB_EVENT": "deployment"},
+    "deployment_status": {"HTTP_X_GITHUB_EVENT": "deployment_status"},
+    "edited_pull_request": pull_request_headers,
+    "fork": {"HTTP_X_GITHUB_EVENT": "fork"},
+    "issue": {"HTTP_X_GITHUB_EVENT": "issues"},
+    "issue_comment": {"HTTP_X_GITHUB_EVENT": "issue_comment"},
+    "issue_comment_deleted": {"HTTP_X_GITHUB_EVENT": "issue_comment"},
+    "labeled_pull_request": pull_request_headers,
+    "member": {"HTTP_X_GITHUB_EVENT": "member"},
+    "membership": {"HTTP_X_GITHUB_EVENT": "membership"},
```

Comment:
So, if you look at this file, almost all of the entries are basically mapping `X` -> `X`, where `X` is something like `membership` or `issue_comment`.

This is the reason I think we should be doing headers via a convention, at least for the GitHub integration, like this:

```
def fixture_filename_to_headers(filename: str) -> Dict[str, Any]:
    if '__' in filename:
        event_type = filename.split("__")[0]
    else:
        event_type = filename
    return {"HTTP_X_GITHUB_EVENT": event_type}
```

Then we just rename the ~10 GitHub fixture files that need this convention, e.g. `merged_pull_request.json` -> `pull_request__merged.json`, and once that's done, there's no longer a magic semi-duplicated list in `headers.py` that needs to be updated every time we add a new fixture.

Reply:
This works pretty well and I like the idea. But we'll have make the system require that `fixture_filename_to_headers` be a method which returns a dictionary instead of being a dictionary itself, which seems pretty reasonable. I'll make the change.